src/java.base/share/classes/javax/crypto/spec/HKDFParameterSpec.java

```diff
@@ -146,21 +149,21 @@ public ExtractThenExpand thenExpand(byte[] info, int length) {
         }
 
         /**
-         * Adds input key material to the builder.
+         * Adds input keying material to the builder.
          * <p>
-         * {@code addIKM} may be called when the input key material value is to
+         * {@code addIKM} may be called when the input keying material value is to
          * be assembled piece-meal or if part of the IKM is to be supplied by a
          * hardware crypto device. This method appends to the existing list of
          * values or creates a new list if there is none yet.
          * <p>
          * This supports the use-case where a label can be applied to the IKM
          * but the actual value of the IKM is not yet available.
-         * <p>
-         * An implementation should concatenate the input key materials into a
-         * single value once all components are available.
+         *
+         * @implNote An implementation should concatenate the input keying
+         * materials into a single value once all components are available.
```

Comment:
I'm not sure what "once all components are available" means. Technically, it's "...into a single value and pass it to the HKDF-Extract step".

Also, this method is for end users. The `@implNote` should be added to `ikms` which is called by an implementation. There you can tell the implementation to do the concatenation.